src/main/java/frc/robot/swerve/ModuleIO.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.swerve;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj2.command.Command;
+
+public interface ModuleIO {
+    void updateInputs(SwerveModuleInputs inputs);
+
+    Rotation2d getAngle();
+
+    void setAngle(Rotation2d angle);
+
+    double getVelocity();
+
+    void setVelocity(double velocity);
+
+    SwerveModuleState getModuleState();
+
+    default SwerveModulePosition getModulePosition(){
+        return null;
+    }
+
+    default void updateOffset(double offset) {
+    }
+
+    default void stopMotor() {
+    }
+
+    default boolean encoderConnected() {
+        return false;
+    }
+
+    default Command checkModule() {
+        return null;
+    }
```

Comment:
I agree that getModulePosition and stopMotor shouldn't but the others shouldn't do anything in the sim IO.

Reply:
OK, makes sense. Then I'd only change the default so that the encoder is always connected, and `checkModule()` should return (`Commands.none()`)[https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/Commands.html#none()], so on the rare case it's used in simulation/replay, the code won't crash because the command scheduler will attempt to run a `null` command.

Reply:
de8e3c6522e2d90e9df4b7580182bdaf686f870e
7c2b3ccf79b0afcaa9fb9b3f8a2f06b9d4dbce62